awsagentprovider/src/main/java/software/amazon/opentelemetry/javaagent/providers/AwsMetricAttributeGenerator.java

```diff
@@ -144,13 +135,27 @@ private static void setRemoteTarget(SpanData span, AttributesBuilder builder) {
    */
   private static Optional<String> getRemoteTarget(SpanData span) {
     if (isKeyPresent(span, AWS_BUCKET_NAME)) {
-      return Optional.ofNullable(span.getAttributes().get(AWS_BUCKET_NAME));
-    } else if (isKeyPresent(span, AWS_QUEUE_NAME)) {
-      return Optional.ofNullable(span.getAttributes().get(AWS_QUEUE_NAME));
-    } else if (isKeyPresent(span, AWS_STREAM_NAME)) {
-      return Optional.ofNullable(span.getAttributes().get(AWS_STREAM_NAME));
-    } else if (isKeyPresent(span, AWS_TABLE_NAME)) {
-      return Optional.ofNullable(span.getAttributes().get(AWS_TABLE_NAME));
+      return Optional.ofNullable("::s3:::" + span.getAttributes().get(AWS_BUCKET_NAME));
+    }
+
+    if (isKeyPresent(span, AWS_QUEUE_URL)) {
+      String arn = SqsUrlParser.getSqsRemoteTarget(span.getAttributes().get(AWS_QUEUE_URL));
+
+      if (arn != null) {
+        return Optional.ofNullable(arn);
+      }
+    }
+
+    if (isKeyPresent(span, AWS_QUEUE_NAME)) {
+      return Optional.ofNullable("::sqs:::" + span.getAttributes().get(AWS_QUEUE_NAME));
+    }
```

Comment:
are these two redundant for SQS?  Will we get both SQS ARN and SQS queue name?

Reply:
We cannot get both. If the first succeeds, then it exits. We try to use the URL first because it has more data potentially.